packages/gatsby-telemetry/src/telemetry.js

```diff
@@ -19,11 +19,37 @@ module.exports = class AnalyticsTracker {
   constructor() {
     try {
       this.componentVersion = require(`../package.json`).version
+      this.installedGatsbyVersion = this.getGatsbyVersion()
+      this.gatsbyCliVersion = this.getGatsbyCliVersion()
     } catch (e) {
       // ignore
     }
   }
 
+  getGatsbyVersion() {
+    const packageInfo = require(join(
```

Comment:
@pieh mentioned https://github.com/sindresorhus/resolve-cwd

Could maybe use that to simplify a lot of this

Reply:
This was mentioned just so same logic as figuring out the `command` file path is used for package.json ( like in https://github.com/gatsbyjs/gatsby/blob/6f28154488cccf998972f386a13a1c1858167a31/packages/gatsby-cli/src/create-cli.js#L74 ). As this can act weirdly in some scenarios (like monorepo or Yarn PnP)

Reply:
yeah resolveCWD looks great I was going to say require.resolve with a try catch but this is way better!

Reply:
I would say this is tangential to this PR so if anything this can be handled in separate one

Reply:
That's exactly how `gatsby` does it itself, should I change it here though?

Reply:
no lets do another PR that fixes all of them